rplugin/python3/denite/source/grep.py

```diff
@@ -40,23 +52,26 @@ def __init__(self, vim):
             'command': ['grep'],
             'default_opts': ['-inH'],
             'recursive_opts': ['-r'],
+            'pattern_opt': ['-e'],
             'separator': ['--'],
             'final_opts': ['.'],
         }
         self.matchers = ['matcher_ignore_globs', 'matcher_regexp']
 
     def on_init(self, context):
         self.__proc = None
-        directory = ''
-        if context['args']:
-            directory = context['args'][0]
-        if not directory:
-            directory = context['path']
-        context['__arguments'] = context['args'][1:]
-        context['__directory'] = self.vim.call('expand', directory)
-        context['__input'] = context['input']
-        if not context['__input']:
-            context['__input'] = self.vim.call('input', 'Pattern: ')
+        try:
+            context['__arguments'] = context['args'][0]
```

Comment:
It is very ugly.  I will convert `context['args']` to Dictinary.
Please use `get()`.